src/AzureDevOpsKubeService.ts

```diff
@@ -34,7 +37,26 @@ export class AzureDevOpsKubeService extends KubeServiceBase {
                 });
             case KubeResourceType.Pods:
                 return this._dataPromise.then((queryResult) => {
-                    return safeParseJson(queryResult.pods);
+                    let podList = safeParseJson(queryResult.pods);
+                    // filter the pods to suit to given filter
+                    if (labelSelector) {
+                        // get all label filter and then check each pod to suit to that filter
+                        let pods: any[] = [];
+                        const originalPods = podList.items || [];
+                        const labelDict = this._getLabelDict(labelSelector);
+                        originalPods.forEach(pod => {
+                            if (pod && pod.metadata && pod.metadata.labels) {
+                                const matched = Object.keys(labelDict).every((key) => pod.metadata.labels[key] === labelDict[key]);

```

Comment:
can we make the comparison case insensitive

Reply:
fixed